router/http.go

```diff
@@ -223,14 +223,15 @@ func (h *httpSyncHandler) Remove(id string) error {
 }
 
 func (s *HTTPListener) listenAndServe(started chan<- error) {
-	_, port, err := net.SplitHostPort(s.Addr)
+	var err error
+	s.listener, err = reuseport.NewReusablePortListener("tcp4", s.Addr)
+	started <- err
 	if err != nil {
-		started <- err
 		return
 	}
-	s.listener, err = reuseport.NewReusablePortListener("tcp4", s.Addr)
-	started <- err
+	_, port, err := net.SplitHostPort(s.listener.Addr().String())
```

Comment:
Maybe we don't need to handle this error here and can just get it from `portFromAddr()` as we're doing in the tests. Then in that func, we can just panic if we hit an error.